src/routes/+layout.svelte

```diff
@@ -103,18 +86,44 @@
             tracked.clear();
         }
 
-        // TODO: thejessewinton, the `updated` from `svelte/state` creates an infinite refresh loop on docs references pages!
-        if ($updated && !willUnload && to?.url) {
+        if (updated.current && !willUnload && to?.url) {
             location.href = to.url.href;
         }
     });
 
-    $: if (browser) currentTheme.subscribe((theme) => applyTheme(theme));
-    $: if (browser && $loggedIn) {
-        document.body.dataset.loggedIn = '';
-    }
+    $effect(() => {
+        if (!navigating?.to) {
+            return;
+        }
+
+        const isDocs = navigating.to.route.id?.startsWith('/docs');
+
+        if (isDocs) {
+            if (!document.body.classList.contains(`${$currentTheme}`)) {
+                applyTheme($currentTheme);
+            }
+        } else {
+            applyTheme('dark');
+        }
+    });
+
+    $effect(() => {
+        currentTheme.subscribe((theme) => applyTheme(theme));
+    });
 
-    $: canonicalUrl = page.url.origin.replace(/^https?:\/\/www\./, 'https://') + page.url.pathname;
+    $effect(() => {
+        if ($loggedIn) {
+            document.body.dataset.loggedIn = '';
+        }
+    });
+
+    let canonicalUrl = $state<string>(
+        `${page.url.origin.replace(/^https?:\/\/www\./, 'https://')}${page.url.pathname}`
+    );
+
+    $effect(() => {
+        canonicalUrl = `${page.url.origin.replace(/^https?:\/\/www\./, 'https://')}${page.url.pathname}`;
+    });
```

Comment:
can we not use `$derived` here, `page` from `state` would always be updated, right?